src/legacy/ui/public/courier/search_strategy/default_search_strategy.ts

```diff
@@ -17,65 +17,8 @@
  * under the License.
  */
 
-import { SearchStrategyProvider, SearchStrategySearchParams } from './types';
-import { addSearchStrategy } from './search_strategy_registry';
-import { isDefaultTypeIndexPattern } from './is_default_type_index_pattern';
-import {
-  getSearchParams,
-  getMSearchParams,
-  getPreference,
-  getTimeout,
-} from '../fetch/get_search_params';
-
-export const defaultSearchStrategy: SearchStrategyProvider = {
-  id: 'default',
-
-  search: params => {
-    return params.config.get('courier:batchSearches') ? msearch(params) : search(params);
-  },
-
-  isViable: indexPattern => {
-    return indexPattern && isDefaultTypeIndexPattern(indexPattern);
-  },
-};
-
-function msearch({ searchRequests, es, config, esShardTimeout }: SearchStrategySearchParams) {
-  const inlineRequests = searchRequests.map(({ index, body, search_type: searchType }) => {
-    const inlineHeader = {
-      index: index.title || index,
-      search_type: searchType,
-      ignore_unavailable: true,
-      preference: getPreference(config),
-    };
-    const inlineBody = {
-      ...body,
-      timeout: getTimeout(esShardTimeout),
-    };
-    return `${JSON.stringify(inlineHeader)}\n${JSON.stringify(inlineBody)}`;
-  });
-
-  const searching = es.msearch({
-    ...getMSearchParams(config),
-    body: `${inlineRequests.join('\n')}\n`,
-  });
-  return {
-    searching: searching.then(({ responses }) => responses),
-    abort: searching.abort,
-  };
-}
-
-function search({ searchRequests, es, config, esShardTimeout }: SearchStrategySearchParams) {
-  const abortController = new AbortController();
-  const searchParams = getSearchParams(config, esShardTimeout);
-  const promises = searchRequests.map(({ index, body }) => {
-    const searching = es.search({ index: index.title || index, body, ...searchParams });
-    abortController.signal.addEventListener('abort', searching.abort);
-    return searching.catch(({ response }) => JSON.parse(response));
-  });
-  return {
-    searching: Promise.all(promises),
-    abort: () => abortController.abort(),
-  };
-}
+import { addSearchStrategy, defaultSearchStrategy } from '../index';
 
 addSearchStrategy(defaultSearchStrategy);
```

Comment:
Hmm, should this be happening somewhere in the shim plugin? Or what's the plan here for migration?

Reply:
Good point, so `addSearchStrategy` was being called as a side effect of importing from `ui/courier/search_strategy/default_search_strategy`. And was imported in the default `uiExports`: https://github.com/elastic/kibana/blob/8e9a8a84dccfa7965ce8a22362885e6cdef8b51f/src/legacy/ui/ui_exports/ui_export_defaults.js#L50

I was worried about removing the `uiExport` right now because I didn't want to break anything in the legacy world. So my plan was to "leave this behind", and update the search strategy registry in the new platform to register the defaultSearchStrategy from the data plugin as soon as it is initialized.

However, this reminds me that I totally forgot to expose the search strategies registry from the shim's runtime contract. Rather than adding another thing to this PR (especially since it has already been tested), I'm going to make this update in [the follow up PR](https://github.com/elastic/kibana/pull/53701) which @lizozom is working on.